src/toolchain.rs

```diff
@@ -315,23 +315,48 @@ impl Toolchain {
             RustupAction::Remove => ("remove", "removing"),
         };
 
-        let thing = thing.to_string();
-        let action = action.to_string();
+        let toolchain_name = self.rustup_name();
+        info!("{log_action_ing} {thing} {name} for toolchain {toolchain_name}");
 
         #[cfg(feature = "unstable-toolchain-ci")]
-        if let ToolchainInner::CI { .. } = self.inner {
-            failure::bail!(
-                "{} {} on CI toolchains is not supported yet",
-                log_action_ing,
-                thing
-            );
+        if let ToolchainInner::CI(ci) = &self.inner {
+            if let RustupAction::Remove = action {
+                failure::bail!("removing {thing} on CI toolchains is not supported yet");
```

Comment:
pre-existing, but I would love to call this "component" instead of "thing" to match rustup's nomenclature.

Reply:
Hmm ... rustup's CLI treats targets and components separately so while we certainly could rename `RustupThing` to `RustupComponent` you'd then have `RustupComponent::Target` which seems confusing (`RustupComponent::Component` isn't great either). 

Do you still want me to do this?

Reply:
Ah, I see. No that's ok, fine to leave it as-is.